modules/cache.js

```diff
@@ -96,32 +100,38 @@ Cache.prototype.quit = function (cb) {
  * This function will be triggered on new block, it will clear all cache entires.
  */
 
-Cache.prototype.onNewBlock = function () {
-	if(!self.isReady()) { return; }
-	async.map(['/api/blocks*', '/api/transactions*'], function (pattern) {
+Cache.prototype.onNewBlock = function (block, broadcast, cb) {
+	cb = cb ? cb: function () {};
```

Comment:
We use cb in multiple places based on certain conditions in other functions. So, using the second approach which you suggested. But good advice. Thanks :+1: